### PR TITLE
New keyboard keymap, small keymap updates to mm2 keymap

### PR DESCRIPTION
**Pull request by arkag**
Got a new keyboard, made my own keymap, merge at your leisure.

### CODE REVIEW

File: keyboards/preonic/keymaps/arkag/config.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#pragma once
```
Comment (drashna) [inline keyboards/preonic/keymaps/arkag/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/preonic/keymaps/arkag/config.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+#ifdef AUDIO_ENABLE
+    #define STARTUP_SONG    SONG(STARTUP_SOUND)
+    #define GOODBYE_SONG    SONG(GOODBYE_SOUND)
+    #define AUDIO_CLICKY
+    #define AUDIO_CLICKY_FREQ_RANDOMNESS 1.0f
+    #define MIDI_BASIC
+    #define MUSIC_MASK keycode != KC_NO
```
Comment (drashna) [inline keyboards/preonic/keymaps/arkag/config.h:9]:
```suggestion
#    define STARTUP_SONG    SONG(STARTUP_SOUND)
#    define GOODBYE_SONG    SONG(GOODBYE_SOUND)
#    define AUDIO_CLICKY
#    define AUDIO_CLICKY_FREQ_RANDOMNESS 1.0f
#    define MIDI_BASIC
#    define MUSIC_MASK keycode != KC_NO
```

File: keyboards/preonic/keymaps/arkag/keymap.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    CK_TOGG, AU_TOG,  MU_TOG,  _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    M_USSR,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, BL_TOGG, BL_STEP, BL_INC,  BL_DEC,  BL_BRTG, _______, _______, _______, _______, _______, _______,
+    RGB_M_P, RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, _______, _______, _______,
+    RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______),
+
+};
+
+#ifdef AUDIO_ENABLE
+    bool music_mask_user(uint16_t keycode) {
```
Comment (drashna) [inline keyboards/preonic/keymaps/arkag/keymap.c:69]:
```suggestion
bool music_mask_user(uint16_t keycode) {
```

File: keyboards/preonic/keymaps/arkag/rules.mk [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+NKRO_ENABLE = yes
+BACKLIGHT_ENABLE = no
+AUDIO_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+TAP_DANCE_ENABLE = no
+UNICODE_ENABLE = yes
+LEADER_ENABLE = yes
+
+# save me space!
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/preonic/keymaps/arkag/rules.mk:17]: This does the same, plus some additional stuff. 
```suggestion
LTO_ENABLE = yes
```

File: keyboards/preonic/keymaps/arkag/keymap.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/preonic/keymaps/arkag/keymap.c:17]: Just a heads up, if you remove this include, and include it in your arkag.h file, you won't need some of the other includes in that file (such as the quantum.h and the process_* stuff.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/preonic/keymaps/arkag/keymap.c (hunk 3 above)
Comment (arkag) [reply keyboards/preonic/keymaps/arkag/keymap.c:67]: I don't understand this right now, with the changes I just pushed, si this resolved?

Comment (drashna) [reply keyboards/preonic/keymaps/arkag/keymap.c:67]: It's a formatting/indentation change, that's all.  Nothing major.

**Comment by arkag** [comment]
@drashna Can we get this merged in? got another batch I want to keep in a separate MR.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**